### PR TITLE
UpdatePrice use case modification

### PR DESCRIPTION
**Pull request by DaisyLi233**
Made changes to UpdatePrice use case a bit so that the presenter has an instance variable that implements the view interface(UpdateIview), and the view implements that view interface (in that case, presenter does not depend on view). This use case is following MVC model as well.
  No need to create output data structure and view model data structure (because the type returned by the use case interactor and everything that is gonna to display on the screen is just like string (notification), so no need to create extra classes.
  I think for now, all my left to do is to connect my part with Kenneth's part. Kenneth is constructing an output DS so that after the use logs in, it would create a new class whose type is output DS, which includes a string username and a boolean type to tell if that user is manager or not. I can directly access to that boolean type by getter method and try to make "update price" function limited to manager only.

### CODE REVIEW

**Review by tonyxieuoft — APPROVED** [review]
Looks good, Daisy! The addition of the UpdatePriceIView will help our code fit the clean architecture model of the MVC better.

File: src/main/java/interface_adaptors/update_price/UpdateController.java [hunk 1]
```diff
@@ -14,13 +14,13 @@ public UpdateController(UpdatePriceInputBoundary useCase){
      * The controller packages the data and sends it off to the Update Use Case
      * @param serialNum the serial number of the item
      * @param price the price to update(for manager only)
-     * @return a String for the View  to present
      */
 
-    public String updateItem_price(String serialNum, double price, boolean updatePrice){
+    public void updateItem_price(String serialNum, double price, boolean updatePrice){
```
Comment (tonyxieuoft) [inline src/main/java/interface_adaptors/update_price/UpdateController.java:20]: Good changes here. I realized it doesn't make sense for the controller to return something back to the screen. Instead, it should be the presenter's job to interact indirectly with the screen.

File: src/main/java/interface_adaptors/update_price/UpdatePresenter.java [hunk 2]
```diff
@@ -5,33 +5,41 @@
...
      * @param data contains serial number of the item, the price to update, and if that staff has the
      *                  rights to update price or not.
-     * @return a String value for the View to present
      */
 
     @Override
-    public String prepareSuccess(UpdatePriceInputData data){
-        return "the price of item(s) with serial number " + data.getSerialNum() +
-                "has been updated to" + data.getPrice();
+    public void prepareSuccess(UpdatePriceInputData data){
```
Comment (tonyxieuoft) [inline src/main/java/interface_adaptors/update_price/UpdatePresenter.java:21]: Same here. Great changes!

**Comment by DaisyLi233** [comment]
***Update (important!): Since this pull request is not merged yet, so the changes I've made to my code can be viewed together in this pull request as well. Basically I just wrote 4 tests to test the functionality of my use case (Update Price), and congratulations, they all passed! Through testing I found a logical problem in my use case interactor, I want to remind Tony @tonyxieuoft to take a look at your use case interactor again since our logic is pretty similar.
If you need help with writing tests, you can take a look at my code and I am glad to answer any questions!

**Pull request merged**